0xProject/0x-monorepo
# [WIP] pull OHLCV records from Crypto Compare

## Description



It scrapes OHLCV records from Crypto Compare and saves to postgres. 

There are some constants of interest that we could convert into env variables:
* EARLIEST_BACKFILL_TIME; this is how far back the script will try to backfill in the absence of any previous data. It's hardcoded to Sept 2010 for now. Ideally we would run once with that, then set it to something more recent. If there are any pairs that don't have data on CC, the script will always try to fetch from this date.

* MAX_CONCURRENT_REQUESTS; this is hardcoded to 18. Should tweak to make better use of the CC rate limit

* TO_CURRENCIES: fiat/major currencies to compare tokens with. It's ETH, USD, and EUR right now.

Workflow:
1. Query events table for token addresses
2. Query Crypto Compare for token symbols
3. Generate trading pairs as {from: token, to: fiat}
4. Fetch latest scraped time for all pairs or set to EARLIEST_BACKFILL_TIME
5. For each pair, fetch from Crypto Compare and save in chronologically ascending order

## Testing instructions



There are some unit tests but I would like to add more test coverage. 
You can build and run (in pipeline package) 
```
$ node lib/src/scripts/pull_ohlcv_cryptocompare.js
```

Then check your postgres db for data. Recommend setting EARLIEST_BACKFILL_DATE to something earlier, e.g. a few months back. Run twice to test that backfill functionality works as expected.

## Types of changes





* New feature (non-breaking change which adds functionality)



## Checklist:
